AKCOO/Scene/ExpenseScene/View/budgetCategorySettingView.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  budgetCategorySettingView.swift
+//  AKCOO
+//
+//  Created by 이연정 on 11/9/24.
+//
+
+import UIKit
+
+class BudgetCategorySettingView: UIView {
+  
+  // MARK: - Views
+  private let categoryLabel = UILabel().set {
+    $0.text = "카테고리"
+    $0.font = UIFont.akFont(.gmarketMedium16)
+    $0.adjustsFontForContentSizeCategory = true
+    $0.textColor = UIColor.akColor(.gray3)
+  }
+  
+  private let buttonStackView = UIStackView().set {
+    $0.axis = .horizontal
+    $0.distribution = .equalSpacing
+    $0.alignment = .center
+  }
+  
+  private let separatorLine =  UIView().set {
+    $0.backgroundColor = UIColor.akColor(.black)
+  }
+  
+  private lazy var buttons: [UIButton] = (0..<5).map { index in
+    UIButton().set {
+      $0.backgroundColor = UIColor.akColor(.gray1)
+      $0.titleLabel?.font = UIFont.akFont(.gmarketMedium14)
+      $0.titleLabel?.adjustsFontForContentSizeCategory = true
+      $0.setContentHuggingPriority(.defaultHigh, for: .horizontal)
+      $0.setContentCompressionResistancePriority(.defaultHigh, for: .horizontal)
+      $0.contentEdgeInsets = UIEdgeInsets(top: 6, left: 11, bottom: 6, right: 11)
+      
+      $0.layer.masksToBounds = true
+      $0.layer.cornerCurve = .continuous
+      
+      $0.setTitle(buttonTitles[index], for: .normal)
+      $0.setTitleColor(.black, for: .normal)
+      $0.tag = index
+      $0.addTarget(self, action: #selector(buttonTapped(_:)), for: .touchUpInside)
+    }
+  }
+  // MARK: - Properties
+  private let buttonTitles = ["교통", "관광", "식비", "쇼핑", "기타"]
+  private let buttonColors: [UIColor] = [.akColor(.akBlue), .akColor(.akGreen), .akColor(.akYellow), .akColor(.akRed), .akColor(.akPurple)]
+  
+  private var selectedButtonIndex: Int? = nil
+  
+  // MARK: - Initializers
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    setupView()
+  }
+  
+  required init?(coder: NSCoder) {
+    super.init(coder: coder)
+    setupView()
+  }
+  
+  // MARK: - Lifecycle
+  override func layoutSubviews() {
+    super.layoutSubviews()
+    updateButtonCornerRadius()
+  }
+  
+  // MARK: - Setup Methods
+  private func setupView() {
+    backgroundColor = .white
+    
+    addSubview(separatorLine)
+    addSubview(categoryLabel)
+    addSubview(buttonStackView)
+    
+    buttons.forEach { buttonStackView.addArrangedSubview($0) }
```

Comment:
buttonStackView에 모든 뷰들이 다 올라간 뒤 
buttonStackView의 위치를 정하는 게 좋아 보입니다! 
작은 화면 먼저 완성하면 어떨까요? 

```suggestion swift 
    buttons.forEach { buttonStackView.addArrangedSubview($0) }
    addSubview(separatorLine)
    addSubview(categoryLabel)
    addSubview(buttonStackView)
```

Reply:
이 부분은!!!!!!!! 제가 무지했습니다 🥹
일단 정답이 있는 줄은 모르지만, UIKit은 뷰의 상위부터 하위를 구성해 나가는 방식이기 때문에 몽쉘이 이전에 작업해 주셨던 것 처럼 작업해 주시는 것이 더 좋아보여요!   
최근에 알게 되어 댓글 남깁니다 ㅎㅎ 

현 우리  프로젝트에 해당 내용을 반영하려 합니다. 몽쉘과 나누었던 대화 덕분에 부모로부터 위치를 잡는 자식을 먼저 집어넣는 게 어색할 수 있음을 깨닫게 되었어요!    

물론, 우리는 setupView와 constraints 코드가 분리되어 (순서가 고정되어) 문제가 생길일은 없지만...    
 남겨봅니다! ㅎㅎ 
 
 최근 PR에 해당 내용을 남겨두었는데 태그할게요!

Reply:
상위부터 하위를 구성해 나가는 방식인지는 사실 알지 못했기에 보노의 첫 코멘트에도 그렇겠다고 생각했었는데 나중에라도 알려주셔서 감사해요!!! 유킷은 상위부터 하위를 구성한다! 기억하겠습니다